lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { act } from 'react-dom/test-utils';
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import 'jest-enzyme';
-import { faCaretDown, faTrash } from '@fortawesome/free-solid-svg-icons';
 
-import NxDropdown, { Props } from '../NxDropdown';
-import NxButton from '../../NxButton/NxButton';
+import { render, screen, fireEvent, createEvent } from '@testing-library/react';
+import { rtlRenderElement, rtlRender, userEvent } from '../../../__testutils__/rtlUtils';
+
 import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
-import NxOverflowTooltip from '../../NxTooltip/NxOverflowTooltip';
-import NxDropdownMenu from '../../NxDropdownMenu/NxDropdownMenu';
-import AbstractDropdown from '../AbstractDropdown';
-import { NxDropdownDivider } from '../../../';
-import { mount } from 'enzyme';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import NxDropdown, { Props } from '../NxDropdown';
 
 describe('NxDropdown', () => {
-  let container: HTMLDivElement | null;
-
   const minimalProps = {
-        label: 'dropdown',
-        isOpen: false,
-        onToggleCollapse: () => {}
-      },
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxDropdown, minimalProps),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxDropdown, minimalProps);
-
-  beforeEach(function() {
-    // Avoid rendering directly on the body.
-    container = document.createElement('div');
-    document.body.appendChild(container);
-  });
+    label: 'dropdown-toggle',
+    isOpen: false,
+    onToggleCollapse: () => {}
+  };
 
-  afterEach(function() {
-    if (container) {
-      document.body.removeChild(container);
-      container = null;
-    }
-  });
-
-  it('renders a button with the appropriate classes and type=button', function() {
-    const component = getMountedComponent();
-    const dropdown = component.find(AbstractDropdown);
-    const button = dropdown.find(NxButton);
-    const icon = button.find(NxFontAwesomeIcon);
-
-    expect(dropdown).toHaveClassName('.nx-dropdown');
-    expect(button).toHaveClassName('.nx-dropdown__toggle');
-    expect(button).toHaveProp('variant', 'tertiary');
-    expect(button).toHaveProp('type', 'button');
-    expect(button.childAt(0)).toContainReact(<span className="nx-dropdown__toggle-label">dropdown</span>);
-    expect(icon).toHaveProp('icon', faCaretDown);
-  });
+  const quickRender = rtlRender(NxDropdown, minimalProps);
+  const renderEl = rtlRenderElement(NxDropdown, minimalProps);
 
   it('renders the button according to the supplied variant', function() {
```

Comment:
Styling classes are implementation details.  The external behavior to test would be the actual styling, tested in visual tests.  If there aren't yet examples and visual tests of each of these variants, they should be added and this test removed.

Reply:
> Styling classes are implementation details. The external behavior to test would be the actual styling, tested in visual tests. If there aren't yet examples and visual tests of each of these variants, they should be added and this test removed.

Removed variants test here: 5762a3468ac606409044e5c26218ec34dacf63a3